plugins/compile.sh

```diff
@@ -7,18 +7,268 @@
+			echo -e "Team Fortress Classic\n"
+			gamemod="tfc"
+			;;
+		7 )
+			echo -e "The Specialists\n"
+			gamemod="ts"
+			;;
+		*)
+	esac
+	
+	echo -ne "##                   (10%)\r"
+	
+	get_base
+	get_game_files $id
+}
+
+get_game () {
+	echo "Please select the Game"
+	echo "  1) Counter-Strike"
+	echo "  2) Day of Defeat"
+	echo "  3) Earth's Special Forces"
+	echo "  4) Half-Life"
+	echo "  5) Natural Selection"
+	echo "  6) Team Fortress Classic"
+	echo "  7) The Specialists"
+	echo "  0) Exit"
+	
+	latest=$(wget -qO- https://www.amxmodx.org/)
+	if [ $? -ne 0 ]
+	then
+		echo "Internet connection is required"
+		quit_script
+	fi
+	
+	echo -n "Number: "
+	read -r id
+	
+	if (( id == 0 )) || (( id > 7 ))
+	then
+		quit_script
+	fi
+	
+	echo -e "\nAMX MOD X Version"
+	echo "  1) 1.9"
+	echo "  2) 1.10"
+	echo -n "Version: "
+	read -r ver
+	
+	if (( ver == 1 ))
+	then
+		version="1.9"
+	elif (( ver == 2 ))
+	then
+		version="1.10"
+	else
+		quit_script
+	fi
+	
+	echo -e "\nBefore updating files, note that all modifications"
+	echo "  of default files will be reset"
+	echo "Default files are shown below"
+	if (( ver == 1 ))
+	then
+		echo "      configs   / hamdata.ini"
+	fi
+	echo "      data      / all default files"
+	echo "      dlls      / all default files"
+	echo "      modules   / all default files"
+	echo "      plugins   / all default files"
+	echo "      scripting / all default files"
+	echo -en "\nContinue ? [Y/n] : "
+	read -r answer
+	
+	if [ ! "$answer" == "Y" ]
+	then
+		quit_script
+	fi
+}
+
+check_for_updates () {
+	check_if_gameserver
+	get_game
+	update_game $id
+	echo -e "\nSucessfully updated"
+	mv $dir/scripting/compile.sh compile.sh
+	rm -rf update_files
+	exit 1
+}
+	
+
+if [ $# != 0 ]
+	then
+		case $1 in
+			"--help" | "-h" )
+				echo "  Usage:"
+				echo "--help    (-h) : This menu"
+				echo "--version (-v) : AMX MOD X Compiler version"
+				echo "--latest  (-l) : Latest AMX MOD X version"
+				echo -e "--update  (-u) : Update files\n"
+				echo "To compile all source files in the directory"
+				echo "  execute the script without any parameters"
+				echo "To compile specific or multiple source files"
+				echo "  execute the script with parameters set as source files' name"
+				echo "  e.g ./compile.sh admin.sma admincmd.sma adminchat.sma"
+				;;
+			 "--version" | "-v" )
+			 	./amxxpc -q | head -n 3
+	 			;;
+	 		"--latest" | "-l" )
+				latest=$(wget -qO- https://www.amxmodx.org/amxxdrop/1.9/amxmodx-latest-base-$os)
+				
+				if [ $? -eq 0 ]
+				then
+					echo "Latest AMX MOD X version"
+					echo -n "1.9 Build: "
+					echo $latest | grep -o -P '(?<=git).*(?=-base)'
+					echo -n "1.10 Build: "
+					wget -qO- https://www.amxmodx.org/amxxdrop/1.10/amxmodx-latest-base-$os | grep -o -P '(?<=git).*(?=-base)'
+				else
+					echo "Internet connection is required"
+				fi
+				;;
+			"--update" | "-u" )
+				check_for_updates
+				;;	
+			*)
+		esac
+		
+		for src in "$@"
+		do
+			if [[ $src == *.sma ]]
```

Comment:
Should we warn in case of wrong extension?

Reply:
Yes, I thought about it the other day and also adding counter to show how many did plugins did compile and how many didn't, if amxxpc returns any information.